core/embed/unix/drbg.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __UNIX_DRBG_H__
+#define __UNIX_DRBG_H__
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "assert.h"
+#include "chacha_drbg.h"
+
+#define DRBG_INIT_NONCE_LENGTH 0
+#define DRBG_INIT_ENTROPY_LENGTH 50
+_Static_assert(CHACHA_DRBG_DERIVATION_FUNCTION_BLOCK_LENGTH -
+                       CHACHA_DRBG_DERIVATION_FUNCTION_PREFIX_LENGTH -
+                       CHACHA_DRBG_DERIVATION_FUNCTION_PADDING ==
+                   DRBG_INIT_ENTROPY_LENGTH + DRBG_INIT_NONCE_LENGTH,
+               "");
+// Make sure entropy in chacha_drbg derivation function fills exactly one block
+// of hashing function. This is not needed it's just an optimalization.
+
+#define DRBG_RESEED_ENTROPY_LENGTH 50
+_Static_assert(CHACHA_DRBG_DERIVATION_FUNCTION_BLOCK_LENGTH -
+                       CHACHA_DRBG_DERIVATION_FUNCTION_PREFIX_LENGTH -
+                       CHACHA_DRBG_DERIVATION_FUNCTION_PADDING ==
+                   DRBG_RESEED_ENTROPY_LENGTH,
+               "");
+
+#define DRBG_RESEED_INTERVAL_CALLS 1024
+
+void drbg_init(const uint8_t *nonce, size_t nonce_length);
```

Comment:
Yes, you're right. I wanted to call `drbg_init` with the nonce set to a reboot counter. Then I found out that probably the best way how to implement it is to have the boot counter in the storage, but the storage is initialised after drbg. So there is probably no reason to expose `nonce` anymore.

Fixed in https://github.com/trezor/trezor-firmware/commit/654f15fda1d77b0b674503a01bc31dadc9156ec6.

But what do you thing about the reboot counter? It's purpose is to prevent from (for example) reusing nonces (even though we are using only deterministic nonces now).

Reply:
> A reboot counter looks like something that is nice to have, but not worth the effort of doing some hacks before the storage is initialized.

There is no need to do any hacks. The most simple solution is to mix it into the drbg just after the storage is initialized.

Reply:
> mix it into the drbg just after the storage is initialized.

Sounds good.